cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java

```diff
@@ -327,9 +339,6 @@ public void run() {
                     catch (URISyntaxException e) {
                         break;
                     }
-                    catch (NullPointerException e) {
-                        break;
```

Comment:
Why not to catch NullPointerException?

Reply:
this is sync the changes from cocos2d-x repo, I don't know this clearly yet. and I don't think it's a good behavior to do nothing after catch a Exception. we should print warn-level log at least.

Reply:
Yep, log output is better. But if we don't know the exact reason of removing `catch Null Pointer Exception` line, let's revert it at least. 
I agree that adding more detailed log output in NullPointerException catching block.

Reply:
thanks for advice, I will research the reason

Reply:
See the comments, `NullPointerException` will throw  when the URL is null
```java
     * @throws  NullPointerException
     *          If {@code str} is {@code null}
    public URI(String str) throws URISyntaxException {
        new Parser(str).parse(false);
    }
```
in  other side, before call those java code, the native code checked the URL
```C++
    std::shared_ptr<const DownloadTask> Downloader::createDownloadFileTask(const std::string& srcUrl,
...
    {
        do
        {
            ...
            if (0 == srcUrl.length() || 0 == storagePath.length())
            {
                if (onTaskError)
                {
                    onTaskError(*task, DownloadTask::ERROR_INVALID_PARAMS, 0, "URL or storage path is empty.");
                }
                task.reset();
                break;
            }
...
```
So, the `NullPointerException` will never occur.